src/eureka/lib/gaussian_min.py

```diff
@@ -0,0 +1,210 @@
+        Written and implemented by Isaac Edelman, edelman@baeri.org
+    """
+    # Create median frame from Integration
+    median_Frame = np.ma.median(img, axis=0)
+
+    # Create initial centroid guess using specified method
+    if meta.centroid_tech.lower() == 'com': 
+        x, y = centroid_com(median_Frame)
+
+    elif meta.centroid_tech.lower() == 'gauss1d':
+        x, y = centroid_1dg(median_Frame)
+
+    elif meta.centroid_tech.lower() == 'gauss2d':
+        x, y = centroid_2dg(median_Frame)
+
+    # Returns x,y coordinates of centroid position
+    return x, y
+
+
+# Second centroid fit for x,y values 
+# + guassian fit for xs, ys
+def mingauss(img, yxguess, meta):
+    """
+    Use an inital centroid guess 
+    and a minimized gaussian meshgrid on each frame 
+    to create an enchanced centroid position (x, y) 
+    and gaussian widths (sx, sy).
+    
+    Parameters
+    ----------
+    img : 2D ndarray
+        Array containing the star image.
+    yxguess : tuple
+        A guess at the centroid. 
+        Defaults to None which uses the data point 
+        with the highest value.
+    meta : eureka.lib.readECF.MetaClass
+        The metadata object.
+
+    Returns
+    -------
+    sy : gaussian width in y direction, float.
+    sx : gaussian width in x direction, float.
+    x : Refined centroid x position, float.
+    y : Refined centroid y position, float.
+
+    Notes
+    -----
+    History:
+
+    - Feb 22, 2023 Isaac Edelman 
+        Written and implemented by Isaac Edelman, 
+        edelman@baeri.org
+    """
+    # Create centroid position x,y 
+    # based off of centroid method 
+    # and inital centroid guess
+    if meta.centroid_tech.lower() == 'com':
+        x, y = centroid_sources(img, yxguess[1], 
+                                yxguess[0], centroid_func=centroid_com, 
+                                box_size=meta.ctr_cutout_size)
+    elif meta.centroid_tech.lower() == 'gauss1d':
+        x, y = centroid_sources(img, yxguess[1], 
+                                yxguess[0], centroid_func=centroid_1dg, 
+                                box_size=meta.ctr_cutout_size)
+    elif meta.centroid_tech.lower() == 'gauss2d':
+        x, y = centroid_sources(img, yxguess[1], 
+                                yxguess[0], centroid_func=centroid_1dg, 
+                                box_size=meta.ctr_cutout_size)
+    
+    # Cropping frame to speed up guassian fit
+    minx = -int(meta.gauss_frame)+int(x)
+    maxx = int(meta.gauss_frame)+int(x)
+
+    # Set Frame size based off of frame crop
+    frame = img[:, minx:maxx]
+
+    # Create meshgrid
+    x_shape = np.arange(img.shape[1])
+    y_shape = np.arange(img.shape[0])
+    x_mesh, y_mesh = np.meshgrid(x_shape, y_shape)
+
+    # Fit the gaussian width by minimizing minfunc with Nelder-Mead method. 
+    # The inital guess is [400, 41, 41] 
+    # or [brightness of pixles in area of interest, xsigma, ysigma].
+    results = minimize(minfunc, [400, 41, 41], 
```

Comment:
Can you add an if/else statement above that uses these values as the initial guess if `meta.inst == 'nircam'` and instead uses `[400, 5, 5]` if `meta.inst == 'miri'`. To avoid duplicated code, make a variable called `guess` that equals these initial guesses and then pass that variable into this function. Also update the comment above to be more general